src/zenml/cli/utils.py

```diff
@@ -555,6 +558,83 @@ def expand_argument_value_from_file(name: str, value: str) -> str:
         )
 
 
+def convert_str_to_dict(string) -> dict:
+    """
+    Converts String into dict type
+
+    Examples:
+        >>> convert_str_to_dict('{"location": "Nevada", "aliens":"many"}')
+        {'location': 'Nevada', 'aliens': 'many'}
+        >>> convert_str_to_dict('location: Nevada \\naliens: many')
+        {'location': 'Nevada', 'aliens': 'many'}
+        >>> convert_str_to_dict("{'location': 'Nevada', 'aliens': 'many'}")
+        {'location': 'Nevada', 'aliens': 'many'}
+
+    Args:
+        string: Json/Yaml/Dict string value
+
+    Returns:
+        dict_: Returns dict from str -> Json/Yaml/dict -> dict
+    """
+    try:
+        dict_ = ast.literal_eval(string)
+        if isinstance(dict_, dict):
+            return dict_
+    except (ValueError, TypeError, SyntaxError):
+        pass
+
+    try:
+        dict_ = json.loads(string)
+        return dict_
+    except ValueError:
+        pass
+
+    try:
+        dict_ = yaml.safe_load(string)
+        return dict_
+    except yaml.YAMLError:
+        pass
+
+    error(f"Invalid argument: '{string}'. Please provide the value in Json/Dict/Yaml format")
+
+
+def read_args_from_file(value: str) -> str:
```

Comment:
Below is the code that is not required for new secret format, so either i can get a logic and fit in with that old function itself to avoid as i cant remove this logic as its used by other functions as well.
https://github.com/zenml-io/zenml/blob/e055d318b326cdc305301d5130b2f2fa46d613e0/src/zenml/cli/utils.py#LL551C1-L554C21

Reply:
If you take my suggestion and allow `--values=@path/to/file`, then this code will also required for the new secret format.